stl/src/atomic_wait.cpp

```diff
@@ -0,0 +1,255 @@
+        // Size is largest of lock-free to make aliasing problem into hypothetical
+        std::atomic<unsigned long long> _Counter;
+
+        CONDITION_VARIABLE _Condition = CONDITION_VARIABLE_INIT;
+        SRWLOCK _Lock                 = SRWLOCK_INIT;
+    };
+#pragma warning(pop)
+
+    _Wait_table_entry& _Atomic_wait_table_entry(const void* const _Storage) noexcept {
+        static _Wait_table_entry wait_table[_Wait_table_size];
+        auto index = reinterpret_cast<std::uintptr_t>(_Storage);
+        index ^= index >> (_Wait_table_size_power * 2);
+        index ^= index >> _Wait_table_size_power;
+        return wait_table[index & _Wait_table_index_mask];
+    }
+
+    static constexpr std::size_t _Uninitialized_spin_count = (std::numeric_limits<std::size_t>::max)();
+    static std::atomic<std::size_t> _Atomic_spin_count     = _Uninitialized_spin_count;
+
+    std::size_t _Atomic_init_spin_count() noexcept {
+        std::size_t result = (std::thread::hardware_concurrency() == 1 ? 0 : 10'000) * _Atomic_spin_value_step;
+        _Atomic_spin_count.store(result, std::memory_order_relaxed);
+        // Make sure other thread is likely to get this,
+        // as we've done kernel call for that.
+        std::atomic_thread_fence(std::memory_order_seq_cst);
+        return result;
+    }
+
+
+#if _STL_WIN32_WINNT >= _WIN32_WINNT_WIN8
+    constexpr bool _Have_wait_functions() {
+        return true;
+    }
+#define __crtWaitOnAddress       WaitOnAddress
+#define __crtWakeByAddressSingle WakeByAddressSingle
+#define __crtWakeByAddressAll    WakeByAddressAll
+
+#pragma comment(lib, "Synchronization.lib")
+
+    void _Atomic_wait_fallback(
+        [[maybe_unused]] const void* const _Storage, [[maybe_unused]] unsigned long long& _Wait_context) noexcept {
+        std::terminate();
+    }
+
+    void _Atomic_notify_fallback([[maybe_unused]] const void* const _Storage) noexcept {
+        std::terminate();
+    }
+
+    void _Atomic_unwait_fallback(
+        [[maybe_unused]] const void* const _Storage, [[maybe_unused]] unsigned long long& _Wait_context) noexcept {}
+
+#else // ^^^ _STL_WIN32_WINNT >= _WIN32_WINNT_WIN8 / _STL_WIN32_WINNT < _WIN32_WINNT_WIN8 vvv
+    void _Atomic_wait_fallback(const void* const _Storage, unsigned long long& _Wait_context) noexcept {
+        switch (_Wait_context & _Atomic_wait_phase_mask) {
+        case _Atomic_wait_phase_wait_none: {
+            _Wait_context = _Atomic_wait_phase_wait_locked;
+            auto& entry   = _Atomic_wait_table_entry(_Storage);
+            ::AcquireSRWLockExclusive(&entry._Lock);
+            [[fallthrough]];
+        }
+
+        case _Atomic_wait_phase_wait_locked: {
+            auto& entry = _Atomic_wait_table_entry(_Storage);
+            ::SleepConditionVariableSRW(&entry._Condition, &entry._Lock, INFINITE, 0);
+            // re-check, and still in _Atomic_wait_phase_wait_locked
+            return;

```

Comment:
Spurious return although it might not be because I'm sure @StephanTLavavej is going to say "I want a default no matter what" or something.

Reply:
Then I guess I should remove it, and put back when `default` is introduced.